crates/forge/src/test_filter.rs

```diff
@@ -0,0 +1,454 @@
+use crate::collecting::CompiledTestCrate;
+use test_collector::TestCase;
+
+#[derive(Debug, PartialEq)]
+// Specifies what tests should be included
+pub(crate) struct TestsFilter {
+    // based on name
+    pub name_filter: NameFilter,
+    // based on `#[ignore]` attribute
+    pub ignored_filter: IgnoredFilter,
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum NameFilter {
+    All,
+    Match(String),
+    ExactMatch(String),
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum IgnoredFilter {
+    NotIgnored,
+    Ignored,
+    All,
```

Comment:
Nit, but `All` could be a first option.

Reply:
I prefered `NotIgnored`, since it is a default one

Reply:
Right, it's a default one. Let's not change it then.